test/libyul/yulOptimizerTests/fullSuite/combine_shift_and_and_3.yul

```diff
@@ -0,0 +1,32 @@
+{
+    let x := calldataload(0)
+    let a := shl(12, shr(4, x))
+    let b := shl(4, shr(12, x))
+    let c := shr(12, shl(4, x))
+    let d := shr(4, shl(12, x))
+    let e := shl(150, shr(2, shl(150, x)))
+    sstore(15, x)
+    sstore(16, a)
+    sstore(17, b)
+    sstore(18, c)
+    sstore(19, d)
+    sstore(20, e)
+}
+// ====
+// EVMVersion: >byzantium
+// ----
+// step: fullSuite
+//
+// {
+//     {
+//         let x := calldataload(0)
+//         let _1 := shl(8, x)
```

Comment:
Is this correct?

For `x = 1` the value of  `shl(12, shr(4, x)) = shl(12, 0) = 0`. But `shl(8, x) = 2**8`

Reply:
Ah nevermind, there is an `and`